content-height-util.js

```diff
@@ -0,0 +1,106 @@
+ * - Set it's initial height to the minimum height (1 line height).
+ * - Set the content
+ * - Compute the content height based on the editor's height & scrollheight.
+ * 
+ * If we run the same algorithm on the live editor (where user is editing too), this provides
+ * false result, and user-experience issue too. This falsle result is specifically in the case
+ * when user is erasing the content frequently (backspace key is kept pressed).
+ */
+
+
+/**
+ * Initial width of the TextEditor in pixels.
+ * This width is actually of no use (in practical), because each time the content-height is 
+ * to be computed actual width of the TextEditor is set on this editor first.
+ */
+const INIT_WIDTH = 500;
+
+
+/**
+ * An instance of the Squire editor which is retrieved from the `iframe.contentWindow.editor`.
+ */
+let textEditor;
+
+/**
+ * An instance of the iframe body which is retrieved from the `iframe.contentWindow.body`.
+ */
+let textContent;
+
+/**
+ * A function used to create/initialize the extra text-editor, which is used to compute the
+ * content-height later.
+ * 
+ * It can be invoked multiple times, but actual work is done only on the first invocation
+ * (when text-editor isn't initialized).
+ * 
+ * Usage:
+ * It is invoked by the TextEditor when it's first rendered.
+ * 
+ * @returns {Promise} Resolved when TextEditor is initialized.
+ * 
+ */
+export const init = (iframePath) => {
+  let resolve, reject;
+  let promise = new Promise((res, rej) => { resolve = res, reject = rej; });
+  if (textEditor) {
+    resolve({textEditor, textContent});
+  } else {
+    let iframe = document.createElement('iframe');
+    iframe.addEventListener('load', () => {
+      textEditor = iframe.contentWindow.editor;
+      textContent = iframe.contentDocument.body;
+      textContent.style.width = INIT_WIDTH + 'px';
+      resolve({textEditor, textContent});
+    });
+
+    iframe.style.position = 'fixed';
+    iframe.style.top = '-99999px';
+    iframe.style.left = '-99999px';
+    document.body.appendChild(iframe);
+    iframe.src = iframePath;
+    window.kerikaTextEditorIframe = iframe;
+  }
+
+  return promise;
+}
+
+/**
+ * Identifies the content-height for the given HTML. 
+ * 
+ * @param {*} html HTML content for which content-height is to be identified.
+ * @param {*} width Width of the TextEditor for which content-height is to be identified.
+ *                  It's important configuration, because the content height is properly
+ *                  computed only when the width of the editor is properly set.
+ * @returns {Number} Required content height.
+ */
+export const getContentHeight = (html, width) => {
+  if (!html) {
+    return 0;
+  }
+
+  if (!textEditor) {
+    throw new Error("textEditor isn't yet initialized. Please invoked `init()` before this.");
+  }
+
+  if (!width) {
+    throw new Error("width is mandatory arguments.");
+  }
+
+  textContent.style.width = width + 'px';
+  textEditor.setHTML(html);
```

Comment:
We didn't set any height to the `iframe` or `textEditor`. Then how this logic will work all the time? Doesn't it cause any issue if we use this method first to find the height of the longer text and then to find the height of shorter text?